proxy/request_handler.go

```diff
@@ -229,6 +229,7 @@ func (d *RequestHandler) HandleRequest(r *http.Request, rl *rule.Rule) (session
 					WithField("granted", false).
 					WithField("authentication_handler", a.Handler).
 					WithField("reason_id", "authentication_handler_error").
+					WithField("error_reason", err.(*herodot.DefaultError).ReasonField).
```

Comment:
I'm pretty sure this will panic at some point.

My plan was to upgrade ory/x here because we've much improved the logger in the new versions. It should include the reason field when logging the error now without any additional work.